code/game/machinery/portable_turret.dm

```diff
@@ -511,61 +379,83 @@ Status: []<BR>"},
 		popDown()
 		return
 
+	last_fired = 0
+	if(attacked)
+		attacked--
+
 	var/list/targets = list()		   // list of primary targets
 	var/list/secondarytargets = list() // targets that are least important
 
-	if(src.check_anomalies) // if its set to check for xenos/carps, check for non-mob "crittersssss"(And simple_animals)
-		for(var/mob/living/simple_animal/C in view(7,src))
-			if(C.stat)
+
+
+	if(emagged)
+		for(var/mob/living/L in view(12, src))
+			if(L.flags & INVULNERABLE)
 				continue
-			if(C.flags & INVULNERABLE)
+			if(istype(L, /mob/living/silicon/robot/mommi) || istype(L, /mob/living/silicon/pai)) //mommis are always safe, can't shoot pAIs
```

Comment:
Should use `isMoMMI()`, on a side note: it shouldn't be possible for the mob pAI to ever be captured in a view, since the actually visible object should always be its container item.